packages/zendesk-adapter/src/filters/brands_filter.ts

```diff
@@ -0,0 +1,167 @@
+  mixedChanges        - Brands with categories order and other changes
+  onlyNonOrderChanges - Brands without any categories order changes
+ */
+const sortBrandChanges = (changes: Change<InstanceElement>[]) :
+    {
+      withOrderChanges : Change<InstanceElement>[]
+      onlyNonOrderChanges : Change<InstanceElement>[]
+    } => {
+  const withOrderChanges : Change<InstanceElement>[] = []
+  const onlyNonOrderChanges : Change<InstanceElement>[] = []
+
+  changes.forEach(change => {
+    if (isRemovalChange(change)) {
+      onlyNonOrderChanges.push(change)
+      return
+    }
+    // currently isn't supported because categories can't exist before brand
+    if (isAdditionChange(change)) {
+      onlyNonOrderChanges.push(change)
+      return
+    }
+    const brandChanges = detailedCompare(change.data.before, change.data.after)
+    const hasAnyOrderChanges = brandChanges.some(c =>
+      c.id.createTopLevelParentID().path[0] === CATEGORIES_FIELD)
+
+    if (hasAnyOrderChanges) {
+      withOrderChanges.push(change)
+    } else {
+      onlyNonOrderChanges.push(change)
+    }
+  })
+
+  return { withOrderChanges, onlyNonOrderChanges }
+}
+
+/**
+ * Handle everything related to brands
+ */
+const filterCreator: FilterCreator = ({ client, config }) => ({
+  /** Insert the brand's categories into a field in it */
+  onFetch: async (elements: Element[]) => {
+    const categories = elements.filter(isInstanceElement)
+      .filter(e => e.elemID.typeName === CATEGORY_TYPE_NAME)
+    const brands = elements.filter(isInstanceElement)
+      .filter(e => e.elemID.typeName === BRAND_TYPE_NAME)
+
+    // Insert the categories of that brand to the brand, sorted
+    brands.forEach(brand => {
+      // If the brand doesn't have Guide activated, do nothing
+      if (!brand.value.has_help_center) {
+        return
+      }
+
+      // Lowest position index first, if there is a tie - the newer is first
+      const brandsCategories = _.orderBy(
+        categories.filter(c => c.value.brand === brand.value.id),
+        ['value.position', 'value.created_at'], ['asc', 'desc']
+      )
+
+      brand.value[CATEGORIES_FIELD] = brandsCategories
+        .map(c => new ReferenceExpression(c.elemID, c))
+    })
+  },
+  /** Change the categories positions according to their order in the brand, and deploy the brand */
+  deploy: async (changes: Change<InstanceElement>[]) => {
+    const [brandChanges, leftoverChanges] = _.partition(
+      changes,
+      change => getChangeData(change).elemID.typeName === BRAND_TYPE_NAME,
+    )
+
+    const { withOrderChanges, onlyNonOrderChanges } = sortBrandChanges(brandChanges)
+
+    const orderChangesToApply: Change<InstanceElement>[] = []
+    const orderChangeErrors: Error[] = []
+
+    withOrderChanges.filter(isAdditionOrModificationChange).forEach(brandChange => {
+      const brandValue = brandChange.data.after.value
+
+      if (brandValue.categories.some((c: Value) => !(c instanceof ReferenceExpression))) {
```

Comment:
please use `isReferenceExpression` - then this can be
```suggestion
      if (!brandValue.categories.every(isReferenceExpression)) {
```